### PR TITLE
feat(ci): update ci to latest standards

### PR DESCRIPTION
**Pull request by baby-gnu**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [x] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [x] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Describe the changes you're proposing

I update the travis and kitchen test matrix to latest `template-formula` standard and fix few things for compatibility:

- feat(ci): update travis and kitchen to latest formula standards
- fix(jinja): encode context as json for python2
- fix(inspec): the package name for Arch is openssh

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

**Comment by myii** [comment]
@baby-gnu Thanks for this PR.  I'm going to have to run this through the `ssf-formula` but there's a bit of a backlog there with other PRs that have already been merged in the org.  Besides, I can see a couple of issues but I'll let the review take place with the commit that is automatically created.  Please give me a couple of days for this, thanks.

**Comment by myii** [comment]
@baby-gnu This is the diff generated by the `ssf-formula`: https://github.com/saltstack-formulas/openssh-formula/commit/71fe2f3061ca6aa6533877db16d3d7fb8f60d535.  It would be more appropriate to amend 10dc1346971235d0b6b633b6f5c98cbf62d54087 with this instead of adding this on top.  There is a slight difference in the matrix since I'm reusing one of the [available YAML node anchors](https://github.com/myii/ssf-formula/pull/239/files#diff-9331ce86567ed9f40bda29728e82c97bR2787).

I've also tested this PR against all of the unique platforms and everything is working fine:

* https://travis-ci.org/github/myii/openssh-formula/builds/709907218

**Comment by baby-gnu** [comment]
> This is the diff generated by the `ssf-formula`: [71fe2f3](https://github.com/saltstack-formulas/openssh-formula/commit/71fe2f3061ca6aa6533877db16d3d7fb8f60d535). It would be more appropriate to amend [10dc134](https://github.com/saltstack-formulas/openssh-formula/commit/10dc1346971235d0b6b633b6f5c98cbf62d54087) with this instead of adding this on top. There is a slight difference in the matrix since I'm reusing one of the [available YAML node anchors](https://github.com/myii/ssf-formula/pull/239/files#diff-9331ce86567ed9f40bda29728e82c97bR2787).

Interesting, I copied the [`travis.yml` from template-formula](https://github.com/saltstack-formulas/template-formula/blob/111a20b47d89d275ce4ff5213656d6828acb2760/.travis.yml#L58), so the new matrix in `openssh-formula` should be what users will have when doing a `./bin/convert-formula.sh`.

If I understand your proposed diff, we should remove tests for the version `3000`?

> I've also tested this PR against all of the unique platforms and everything is working fine:

Great to read that.

**Comment by baby-gnu** [comment]
Great, it's working.

File: .travis.yml [hunk 1]
```diff
@@ -54,50 +54,52 @@ jobs:
...
+    #       testing of the formula's behaviour. If possible, try to refrain from
+    #       the classical "chosing all the instances because I want to test on
+    #       another/all distro/s" trap: it will just add time to the testing (see
+    #       the discussion on #121).  As an example, the set chosen below covers
+    #       the most used distros families, systemd and non-systemd and the latest
+    #       three supported Saltstack versions with python2 and 3.
+    #       As for `kitchen.yml`, that should still contain all of the platforms,
+    #       to allow for comprehensive local testing
+    #       Ref: https://github.com/saltstack-formulas/template-formula/issues/118
+    #       Ref: https://github.com/saltstack-formulas/template-formula/issues/121
```
Comment (myii) [inline .travis.yml:71]: We remove this block from all formulas except the `template-formula`.  I've recently opened up a discussion with @dafyddj to confirm that the conversion script should also do this.

Comment (baby-gnu) [reply .travis.yml:71]: ok, sorry

**Comment by myii** [comment]
Merged, excellent work @baby-gnu!

**Pull request merged**